imagebuildah/build.go

```diff
@@ -216,20 +216,19 @@ func BuildDockerfiles(ctx context.Context, store storage.Store, options BuildOpt
 			}
 			data = resp.Body
 		} else {
-			// If the Dockerfile isn't found try prepending the
-			// context directory to it.
 			dinfo, err := os.Stat(dfile)
-			if os.IsNotExist(err) {
-				// If they are "/workDir/Dockerfile" and "/workDir"
-				// so don't joint it
+			if err != nil {
```

Comment:
Do we really want to do this on an error such as permission denied or on a closed file? I think we want to stick with the IsNotExist() check, but you might sway me.

Reply:
Consider the following scenario:

1. A Podman API server is running (`podman system service`) in the `/usr/bin` working directory (it's `os.Getwd() == "/usr/bin"`) with the following contents:

    ```
    lrwxrwxrwx 1 root root          18 Oct 14 07:38 docker
    -rwxr-xr-x 1 root root    42348552 Oct  2 17:17 podman
    ```

2. Podman receives [`/libpod/build`](https://podman.readthedocs.io/en/latest/_static/api.html#operation/libpodBuildImage) request with a `dockerfile=docker/Dockerfile` argument and a Tar-file in it's body containing the following contents:
    ```
    drwxr-xr-x edi/edi           0 2020-11-10 19:59 docker/
    -rw-r--r-- edi/edi           0 2020-11-10 19:59 docker/Dockerfile
    ```

2. Podman extracts the Tar-file into `/tmp` and calls `BuildDockerfiles()`.

3. `BuildDockerfiles()` fails since it first tries to open the file in the current working directory and `os.IsNotExist(err) == false` (because the file actually exists, it's just not a directory), however it could've worked fine if the context directory was prepended.

While more conditions can be added to this `if`, this will be a semantically tough decision since even a permission denied error can be a valid use-case in the scenario above, where access to files other than `/tmp` is restricted by a security subsystem.

Finally, assuming that `os.Stat()` is cheap on modern OS-es, the only downside I see is that the original error might be swallowed, hindering the problem diagnosis.

Reply:
OK, I'm a bit confused by your point 4, and it may be due to sitting in too many meetings today.  IsNotExists doesn't care if the thing being checked for is a file or a directory, it just checks to see if either a file and a directory with that name doesn't exist.  So if the directory isn't there and a file with that name isn't there, then the old line 222 should trigger.  So I think the old code should trigger in your scenario.

My main concern is we're just overprocessing with this change, but if others think it's OK as you have it, then I'm fine.  It would be interesting to know if your new test passes with out your code change.  Did you by chance try that?

Thanks for the patience and explanations.

Reply:
>So if the directory isn't there and a file with that name isn't there, then the old line 222 should trigger. So I think the old code should trigger in your scenario.

It won't trigger because the error is different (`os.ErrNotExist` vs `*os.PathError`), see https://play.golang.org/p/ZNr3fMQQ43d for an example.

>It would be interesting to know if your new test passes with out your code change.

It doesn't.